crypto/provider_predefined.h

```diff
@@ -9,13 +9,15 @@
 
 #include <openssl/core.h>
 
+int ossl_default_provider_init(const OSSL_PROVIDER *provider,
+                               const OSSL_DISPATCH *in,
+                               const OSSL_DISPATCH **out);
```

Comment:
You can actually do this:

``` C
OSSL_provider_init_fn ossl_default_provider_init;
```

Reply:
I rebased on top of your latest version of #8480 - so picked this up from your branch.